analyzers/src/SonarAnalyzer.Common/Helpers/KnownType.cs

```diff
@@ -686,6 +687,7 @@ public sealed partial class KnownType
         public static readonly ImmutableArray<KnownType> RouteAttributes =
             ImmutableArray.Create(
                 Microsoft_AspNetCore_Mvc_RouteAttribute,
-                System_Web_Mvc_RouteAttribute);
+                System_Web_Mvc_RouteAttribute,
+                System_Web_Mvc_RoutePrefixAttribute);
```

Comment:
I don't know if it is a good idea to mix RouteAttribute and RoutePrefixAttribute, but it is okay. Ultimately, both define a route, even though one of them does so only partially.
@antonioaversa What do you think about it?

Reply:
`RoutePrefix` only exists in ASP.NET 4.x, and only applies at the controller level. 
![image](https://github.com/SonarSource/sonar-dotnet/assets/2676742/96d965e2-5053-4180-9e75-98b171b03d42)

In ASP.NET Core, routing is always additive, i.e. the route at action level doesn't override, but is appended to the route template of the controller. 
So you always only use `Route`, both at the controller and at the action level.

Back to S6931: this rule is only concerned with route templates of actions, so I think I would not add it.
@mary-georgiou-sonarsource If you remove the value from the array, do you have any test breaking? Can we define a scenario where the rule would behave differently, depending on whether we add or don't add the `RoutePrefix` to `RouteAttributes`? 

From what I can see, I guess the difference in behavior comes from the message:

> If there is at least a route template on the controller: Change the paths of the actions of this controller to be relative and adapt the controller route accordingly.
>
> If there is no route template on the controller: Change the paths of the actions of this controller to be relative and add a controller route with the common prefix.

In this scenario, `RoutePrefix` and `Route` are indeed equivalent: in ASP.NET 4.x the "\" in the route template would be added via a `RoutePrefix`, instead of a `Route` attribute.

However, there may be other scenarios where `RoutePrefix` and `Route` are not conceptually equivalent, and adding both to the same `RouteAttributes` array may lead to bugs in other rules.
For these reasons, I would probably:
- not add `RoutePrefix` to `RouteAttributes`
- still define `RoutePrefix` in `KnownType`, but as dedicated `KnownType`
- change the code `x.AttributeClass.IsAny(KnownType.RouteAttributes)` in `ReportIssues` to `x.AttributeClass.IsAny(KnownType.RouteAttributes) || x.AttributeClass.Is(KnownType.RoutePrefixAttribute)`

Having said that, I don't have a strong opinion, and I would be OK with keeping `RoutePrefix` in `RouteAttributes`, if you prefer so.

Reply:
Indeed it's only on the controller!
I will remove it from the collection and check only when I do checks for the controller part!
Thanks to both.